### PR TITLE
add sqlalchemy session termination

### PR DESCRIPTION
**Pull request by iporollo**
## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Other (please describe):

## Issue

db connection sessions not terminated and overloading

## Description of change & new behavior

adds termination to db sessions & connections

### CODE REVIEW

**Review by unkrich — APPROVED** [review]
scary without tests

**Comment by unkrich** [comment]
Biggest takeaway here is we may consider with block

File: src/server/pipeline.py [hunk 1]
```diff
@@ -79,11 +79,24 @@ def uniq(arr): # TODO: Transformer should handle uniqueness
         except Exception as e:
             logging.error(e)
 
+    def _terminate(self):
+        if not self.engine and not self.connection:
+            return
```
Comment (unkrich) [inline src/server/pipeline.py:84]: same

**Review by unkrich — APPROVED** [review]

File: src/ingestion/sources/base.py [hunk 2]
```diff
@@ -148,10 +148,24 @@ def _execute(self, sql: str):
 
         return results
 
+    def _terminate(self):
+        if not self.engine and not self.connection:
+            return
+
```
[the remark this answers is not in the document]
Comment (iporollo) [reply src/ingestion/sources/base.py]: removed

File: src/server/pipeline.py (hunk 1 above)
Comment (iporollo) [reply src/server/pipeline.py]: removed

**Review by unkrich — APPROVED** [review]

**Pull request merged**